src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/Editor/DefaultTextBufferProjectService.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Text;
+using MonoDevelop.Ide;
+using MonoDevelop.Ide.TypeSystem;
+using MonoDevelop.Projects;
+
+namespace Microsoft.VisualStudio.Mac.LanguageServices.Razor.Editor
+{
+    /// <summary>
+    /// Infrastructure methods to find project information from an <see cref="ITextBuffer"/>.
+    /// </summary>
+    internal class DefaultTextBufferProjectService : TextBufferProjectService
+    {
+        private const string MvcAssemblyName = "Microsoft.AspNetCore.Mvc.Razor";
+        private const string AspNetCoreWebCapability = "AspNetCore";
+
+        private readonly ITextDocumentFactoryService _documentFactory;
+        private readonly ErrorReporter _errorReporter;
+        private readonly MonoDevelopWorkspace _workspace;
+
+        public DefaultTextBufferProjectService(
+            ITextDocumentFactoryService documentFactory,
+            ErrorReporter errorReporter,
+            MonoDevelopWorkspace workspace)
+        {
+            if (documentFactory == null)
+            {
+                throw new ArgumentNullException(nameof(documentFactory));
+            }
+
+            if (errorReporter == null)
+            {
+                throw new ArgumentNullException(nameof(errorReporter));
+            }
+
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            _documentFactory = documentFactory;
+            _errorReporter = errorReporter;
+            _workspace = workspace;
+        }
+
+        public override object GetHostProject(ITextBuffer textBuffer)
+        {
+            if (textBuffer == null)
+            {
+                throw new ArgumentNullException(nameof(textBuffer));
+            }
+
+            // If there's no document we can't find the FileName, or look for an associated project.
+            if (!_documentFactory.TryGetTextDocument(textBuffer, out var textDocument))
+            {
+                return null;
+            }
+
+            var projectsContainingFilePath = _workspace.MonoDevelopSolution.GetProjectsContainingFile(textDocument.FilePath);
```

Comment:
Today this always returns 0 projects. Will need to wait on the MonoDevelop folks to fill in the Workspace with the right information.

If you want to test this manually you can replace this line with:
`var projectsContainingFilePath = IdeApp.Workspace.GetProjectsContainingFile(textDocument.FilePath);`

Reply:
Ah I see. Yeah I wasn't sure which project service you were talking about. This class shouldn't rely on roslyn. In VS (windows) this uses `IVsHierarchy`.